Tools: uploader.py: verify correct install

rafasaurus opened the pull request:
- Implement functionality to verify correct firmware installation
- Utilize existing code with an additional argument for verification

peterbarker commented on Tools/scripts/uploader.py:
```diff
@@ -1120,6 +1134,7 @@ def main():
     )
     parser.add_argument('--download', action='store_true', default=False, help='download firmware from board')
     parser.add_argument('--identify', action="store_true", help="Do not flash firmware; simply dump information about board")
+    parser.add_argument('--verify', action="store_true", help="Do not flash firmware; simply verify the firmware")
```
```suggestion
    parser.add_argument('--verify', action="store_true", help="Do not flash firmware; verify that the firmware on the board matches the supplied firmware")
```

timtuxworth replied:
How about --validate?

rafasaurus replied:
--verify-installed or perhaps --validate-installation

peterbarker reviewed (changes requested):
"verify" is really, "verify that the flash worked".

You're re-using the functionality, but that terminology isn't really applicable in this case.

I'm OK with the current name of the parameter, but if we can find something better that would be good!

`--assert-firmware-is` would work, but a little verbose

In any case, a slightly better description is called for...

peterbarker reviewed (changes requested):
... also, please fold all of this into the one commit .  The one starting with `Tools:` looks appropriate.

rafasaurus commented:
> ... also, please fold all of this into the one commit . The one starting with `Tools:` looks appropriate.

Should we just squash and merge?

peterbarker commented on Tools/scripts/uploader.py:
```diff
@@ -882,6 +882,20 @@ def board_name_for_board_id(self, board_id):
         return None
 
+    # verify the firmware
+    def verify(self, fw, boot_delay=None):
```
```suggestion
    def verify_firmware_is(self, fw, boot_delay=None):
```

peterbarker commented on Tools/scripts/uploader.py:
```diff
@@ -1120,6 +1134,8 @@ def main():
     )
     parser.add_argument('--download', action='store_true', default=False, help='download firmware from board')
     parser.add_argument('--identify', action="store_true", help="Do not flash firmware; simply dump information about board")
+    parser.add_argument('--verify', action="store_true",
```
```suggestion
    parser.add_argument('--verify-firmware-is', action="store_true",
```

peterbarker reviewed (changes requested):
Please avoid merge commits in your history - this needs a rebase again.

I've also requested a change you suggested yourself.  I think the suggestion makes it unambiguous what the new feature does?

rafasaurus commented:
> Please avoid merge commits in your history - this needs a rebase again.
> 
> I've also requested a change you suggested yourself. I think the suggestion makes it unambiguous what the new feature does?

Changed, rebased and tested on the board.

peterbarker reviewed (approved):
The permissions changed on the modified file from 0755 to 0644.  I've fixed this.

I've tested this locally and it works as expected.

Marking as MergeOnCIPass - thanks!
